Criipto.Signatures/Client.cs

```diff
@@ -183,6 +183,72 @@ public async Task<SignatureOrder> CloseSignatureOrder(string signatureOrderId, C
         return await CloseSignatureOrder(input).ConfigureAwait(false);
     }
 
+    public async Task<SignatureOrder> CancelSignatureOrder(CancelSignatureOrderInput input)
+    {
+        var response = await graphQLClient.SendMutationAsync(
+            CancelSignatureOrderMutation.Request(new { input = input }),
+            () => new { cancelSignatureOrder = new CancelSignatureOrderOutput() }
+        ).ConfigureAwait(false);
+
+        if (response.Errors?.Length > 0)
+        {
+            throw new GraphQLException(response.Errors[0].Message);
+        }
+
+        return response.Data.cancelSignatureOrder.signatureOrder;
+    }
+
+    public async Task<SignatureOrder> CancelSignatureOrder(SignatureOrder signatureOrder)
+    {
+        if (signatureOrder == null) throw new ArgumentNullException(nameof(signatureOrder));
+
+        var input = new CancelSignatureOrderInput();
+        input.signatureOrderId = signatureOrder.id;
+        return await CancelSignatureOrder(input).ConfigureAwait(false);
+    }
+
+    public async Task<SignatureOrder> CancelSignatureOrder(string signatureOrderId)
```

Comment:
@sgryt How do you feel about Partial classes in C#? I think these individual mutations could be good candidates to put in partial classes.

Reply:
Not a great fan of partial classes, they are typically use by RAD-style drag-drop-programming tools.
I mostly consider them a design smell.

Which use cases do you have in mind ?

Reply:
@sgryt Mostly to declutter the `Client.cs` file since we have 3-4 methods defined for each mutation.

Reply:
That seems like something we could handle by pulling out the input parameter variations to mutation-specific builders.

And then perhaps treat the input params as immutable data (instead of the current in-place field mutations).

Reply:
@sgryt I'm not sure I understand how that would solve the need for method overloads?

Reply:
> @mickhansen We could then reduce the number of method overloads to 1 per mutation in `Client.cs`, and leave the details of handling the various parameter combinations to the builders (which could be placed in either 1 file, or a file per builder, which would be idiomatic in C# land).

Builders could then provide constructor overloads as a replacement for the current method overloads, and expose a `Build()` method that returns the mutation-specific input type instance.

Reply:
Nicer API: No, but DX of Intellisense is better when it only has to show 3-4 ctor overloads (compared to showing all the methods on the `Client`).

Blocking: Technically, no, but changing it after releasing v1 will be breaking.

Reply:
@sgryt V1 is already released, so it would be a change for V2

Reply:
> @sgryt V1 is already released, so it would be a change for V2

Right :) If you are OK with continuing the current strategy and then obsoleting it at some later point (should need be), then fine by me.